lib/lndclient/LndClient.ts

```diff
@@ -506,20 +506,28 @@ class LndClient extends SwapClient {
   private executeSendRequest = async (
     request: lndrpc.SendRequest,
   ): Promise<string> => {
+    if (!this.isConnected()) {
+      throw swapErrors.FINAL_PAYMENT_ERROR(errors.UNAVAILABLE(this.currency, this.status).message);
+    }
+
     this.logger.trace(`sending payment with ${JSON.stringify(request.toObject())}`);
     let sendPaymentResponse: lndrpc.SendResponse;
     try {
       sendPaymentResponse = await this.sendPaymentSync(request);
     } catch (err) {
       this.logger.error('got exception from sendPaymentSync', err);
-      throw swapErrors.PAYMENT_ERROR(err.message);
+      if (typeof err.message === 'string' && err.message.includes('chain backend is still syncing')) {
+        throw swapErrors.FINAL_PAYMENT_ERROR(err.message);
+      } else {
+        throw swapErrors.UNKNOWN_PAYMENT_ERROR(err.message);
```

Comment:
This is the worst case. We have no idea if the payment was done or not.

Reply:
Yes, so when we encounter an unknown payment error we fail the swap but we *don't* respond to the resolve request with `retry=false`.